### PR TITLE
fix: refactor batches fetching

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/rnd-dev-team#1800

- convert `addDataToBatches` into a hook
- use pagination for batches everywhere across the app (except for stats on the activity page)
- merge all batches fetching hook into a single one `useFetchPaginatedBatches`
- replace `useBatchesWithSupply` (without react-query) by `useFetchPaginatedBatches` (with react-query)
- add `getBatchesByIssuerQuery` method

__For reviewers:__
In order to look for regressions, I've put links to all pages using credit batches.
Let me know if I missed any.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1. https://deploy-preview-2152--regen-marketplace.netlify.app/ecocredit-batches/1
2. https://deploy-preview-2152--regen-marketplace.netlify.app/profile/credit-batches
3. https://deploy-preview-2152--regen-marketplace.netlify.app/project/C01-001
4. https://deploy-preview-2152--regen-marketplace.netlify.app/credit-classes/C01
5. https://deploy-preview-2152--regen-marketplace.netlify.app/stats/activity

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-marketplace/src/hooks/batches/useBatchesWithMetadata.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { getBatchQuery } from 'lib/queries/react-query/ecocredit/getBatchQuery/g
 
 import { useProjectsWithMetadata } from 'hooks/projects/useProjectsWithMetadata';
 
-export const useBatchesWithMetadata = (
+export const useFetchBatchesWithMetadata = (
```
Comment (blushi) [inline web-marketplace/src/hooks/batches/useBatchesWithMetadata.ts:8]: can we also rename the file?

**Review by blushi — COMMENTED** [review]
On the stats page, for the number of credits at the top, "-" is getting displayed first instead of the loading skeleton
Otherwise looking good

**Comment by blushi** [comment]
It might be worth for @erikalogie to test this so we have another person double checking there's no regression, see testing instructions above, thanks!

**Review by blushi — APPROVED** [review]
Nice refactoring! It would be great to have @erikalogie to test this as well for any regression but tACK on my side

**Comment by erikalogie** [comment]
Is it expected that after you click through a few pages, some of the pages of credit batches are missing the tx hash and/or the credit class?

**Comment by flagrede** [comment]
@erikalogie Yes, I think we already had that on dev.

**Comment by erikalogie** [comment]
Ok then otherwise I believe it looks ok

**Comment by blushi** [comment]
> @erikalogie Yes, I think we already had that on dev.

Yeah I think so too, the missing txs are related to an existing limitation: https://github.com/regen-network/regen-web/pull/2152#discussion_r1350021212 (not reproducable on prod yet because we don't have enough credit batches), but the missing credit class could be worth investigating as part of a separate issue

**Pull request merged**